purchase_delivery_split_date/models/purchase.py

```diff
@@ -132,7 +132,7 @@ def _check_split_pickings(self):
                             continue
                         if (
                             move.picking_id.scheduled_date.date() != date_key
-                            or pickings_by_date[date_key] != move.picking_id
+                            or pickings_by_date.get(date_key, []) != move.picking_id
```

Comment:
Just a nitpick: I find using a list as a default value confusing and I would not be surprised if this comparison was going to give a TypeError at some point. You can just leave it out, because comparing with None is at least just as valid.
```
In [2]: picking
Out[2]: stock.picking(1,)

In [3]: picking != []
Out[3]: True

In [4]: picking != None
Out[4]: True
```